src/img_to_text_gen.py

```diff
@@ -0,0 +1,50 @@
+import torch
+from transformers import VisionEncoderDecoderModel, \
+    ViTFeatureExtractor, AutoTokenizer
+
+
+class IttGenerator(object):
+    def __new__(cls):
+        if not hasattr(cls, 'instance'):
+            cls.instance = super(IttGenerator, cls).__new__(cls)
+        return cls.instance
+
+    def __init__(self):
+        self.model = VisionEncoderDecoderModel\
+            .from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+        self.feature_extractor = ViTFeatureExtractor\
+            .from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+        self.tokenizer = AutoTokenizer\
+            .from_pretrained("nlpconnect/vit-gpt2-image-captioning")
+        self.device = torch\
+            .device("cuda" if torch.cuda.is_available() else "cpu")
+        self.model.to(self.device)
+
+    def gen(self, image, max_length, min_length, num_beams):
+        if image is None:
+            return "Failed wile opening an image"

```

Comment:
while

Reply:
There is no distinction between correct output and error. It is better to raise an exception here, e.g. `ValueError`.